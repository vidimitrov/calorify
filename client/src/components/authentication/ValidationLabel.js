import styled, { css } from 'styled-components';

const ValidationLabel = styled.p`
  ${props => !props.valid && css`
    color: #ff387b;
    text-align: left;
    margin: 0 auto 20px;
    width: 85%;
  `}
`;

export default ValidationLabel;
