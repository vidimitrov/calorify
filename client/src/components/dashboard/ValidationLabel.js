import styled, { css } from 'styled-components';

const ValidationLabel = styled.p`
  display: none;

  ${props => props.invalid && css`
    display: block;
    color: #ff387b;
    text-align: left;
    margin: 0;
    width: 85%;
  `}
`;

export default ValidationLabel;
