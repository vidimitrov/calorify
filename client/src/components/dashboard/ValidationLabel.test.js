import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ValidationLabel from './ValidationLabel';

it('Dashboard/ValidationLabel component renders correctly', () => {
  const tree = renderer.create(
    <ValidationLabel invalid>Validation in not correct</ValidationLabel>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
