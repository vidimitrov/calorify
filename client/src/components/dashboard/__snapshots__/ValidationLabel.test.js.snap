// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard/ValidationLabel component renders correctly 1`] = `
.c0 {
  display: none;
  display: block;
  color: #ff387b;
  text-align: left;
  margin: 0;
  width: 85%;
}

<p
  className="c0"
>
  Validation in not correct
</p>
`;
